src/client/Microsoft.Identity.Client/ApiConfig/AcquireTokenForClientParameterBuilder.cs

```diff
@@ -169,22 +168,7 @@ protected override void Validate()
         {
             if (CommonParameters.MtlsCertificate != null)
             {
-                string authorityUri = ServiceBundle.Config.Authority.AuthorityInfo.CanonicalAuthority.AbsoluteUri;
-
-                if (ServiceBundle.Config.Authority.AuthorityInfo.AuthorityType != AuthorityType.Aad && 
-                    ServiceBundle.Config.Authority.AuthorityInfo.AuthorityType != AuthorityType.Dsts)
-                {
-                    throw new MsalClientException(
-                        MsalError.InvalidAuthorityType,
-                        MsalErrorMessage.MtlsInvalidAuthorityTypeMessage);
-                }
-
-                if (authorityUri.Contains("/common", StringComparison.OrdinalIgnoreCase))
-                {
-                    throw new MsalClientException(
-                        MsalError.MissingTenantedAuthority,
-                        MsalErrorMessage.MtlsNonTenantedAuthorityNotAllowedMessage);
-                }
+                ServiceBundle.Config.Authority.AuthorityInfo.ThrowIfNotSupportedForMtls();
```

Comment:
There's an easier of doing it - just look at `ServiceBundle.Config.Authority.TenantId`

Reply:
Thanks @bgavrilMS for the pointer. After looking at it, I think we have at least the below options. I prefer #2 from below. What do you think?

1) Leave the ``AuthorityInfo.ThrowIfNotSupportedForMtls`` as-is.
2) Move the ``ThrowIfNotSupportedForMtls`` in ``Authority`` as an abstract function and have each authority implement it. 
3) Leave ``AuthorityType`` check here (as before) and only check the ``Tenant`` in the ``AuthorityInfo.ThrowIfNotSupportedForMtls``.
4) Revert the changes and replace the ``/common`` check using the suggested ``ServiceBundle.Config.Authority.TenantId``.

Reply:
Move the `ThrowIfNotSupportedForMtls` in `Authority`